great_expectations/validator/validator.py

```diff
@@ -1923,20 +1923,59 @@ def set_evaluation_parameter(self, parameter_name, parameter_value):
     def add_citation(
         self,
         comment: str,
+        batch_request_list: Optional[
+            List[Union[str, Dict[str, Union[str, Dict[str, Any]]]]]
+        ] = None,
+        batch_request: Optional[
+            Union[str, Dict[str, Union[str, Dict[str, Any]]]]
+        ] = None,
         batch_spec: Optional[dict] = None,
         batch_markers: Optional[dict] = None,
         batch_definition: Optional[dict] = None,
         citation_date: Optional[str] = None,
     ) -> None:
         """Adds a citation to an existing Expectation Suite within the validator"""
-        if batch_spec is None:
+        if batch_spec is None and hasattr(self, "batch_spec"):
             batch_spec = self.batch_spec
-        if batch_markers is None:
+        if batch_markers is None and hasattr(self, "batch_markers"):
             batch_markers = self.active_batch_markers
-        if batch_definition is None:
+        if batch_definition is None and hasattr(self, "batch_definition"):
```

Comment:
@NathanFarmer You may wish to solicit the opinion from @cdkini regarding whether or not using `hasattr` is justified in this case.  Thanks.